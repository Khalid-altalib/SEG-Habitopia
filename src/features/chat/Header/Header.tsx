import { Card, Center, Heading, Text, VStack } from "native-base";
import React from "react";

type Props = {};

const Header = (props: Props) => {
  return (
    <Center>
      <Heading fontSize="4xl">Sleep Challenge</Heading>
      <Text fontSize="xl">Time to catch up on your sleep.</Text>
    </Center>
  );
};

export default Header;
