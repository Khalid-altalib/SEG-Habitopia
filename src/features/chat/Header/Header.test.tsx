import React from "react";
import { render } from "@testing-library/react-native";
import configureStore from "redux-mock-store";
import thunk from "redux-thunk";
import TestingWrapperNavigation from "@app/testingWrapperWithNavigation";
import chatMockState from "../chatMockState";
import Header from "./Header";

describe("Header", () => {
    const mockState = {chat: chatMockState,};
    // create a mock Redux store with the mock state and middleware
    const mockStore = configureStore([thunk])(mockState);
    const Props = {
        name: 'Fitness Chat',
        description: 'Be active',
    };

    it("renders the header component correctly", async () => {
      const component = render(
        <TestingWrapperNavigation store={mockStore}>
          <Header {...Props} />
        </TestingWrapperNavigation>
    );
      // expect the rendered component to match the snapshot
      expect(component).toMatchSnapshot();
    });
  });