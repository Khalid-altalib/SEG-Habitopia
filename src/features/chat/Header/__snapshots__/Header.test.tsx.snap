// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders the header component correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <RNSScreenStack
      style={
        {
          "flex": 1,
        }
      }
    >
      <RNSScreen
        collapsable={false}
        forwardedRef={[Function]}
        gestureResponseDistance={
          {
            "bottom": -1,
            "end": -1,
            "start": -1,
            "top": -1,
          }
        }
        nativeBackButtonDismissalEnabled={false}
        onAppear={[Function]}
        onDisappear={[Function]}
        onDismissed={[Function]}
        onHeaderBackButtonClicked={[Function]}
        onNativeDismissCancelled={[Function]}
        onTransitionProgress={[Function]}
        onWillDisappear={[Function]}
        replaceAnimation="push"
        stackPresentation="push"
        style={
          {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
        swipeDirection="horizontal"
      >
        <RNSScreenStackHeaderConfig
          backButtonInCustomView={false}
          backgroundColor="rgb(255, 255, 255)"
          color="rgb(0, 122, 255)"
          direction="ltr"
          disableBackButtonMenu={false}
          hidden={false}
          hideBackButton={false}
          largeTitleHideShadow={false}
          title="TestScreen"
          titleColor="rgb(28, 28, 30)"
          topInsetEnabled={false}
          translucent={false}
        />
        <View
          accessibilityElementsHidden={false}
          importantForAccessibility="auto"
          style={
            {
              "flex": 1,
              "flexDirection": "column-reverse",
            }
          }
        >
          <View
            style={
              [
                {
                  "flex": 1,
                },
                {
                  "backgroundColor": "rgb(242, 242, 242)",
                },
                undefined,
              ]
            }
          >
            <View
              dataSet={{}}
              style={
                {
                  "alignItems": "center",
                  "display": "flex",
                  "justifyContent": "center",
                  "marginBottom": 20,
                }
              }
            >
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 36,
                    "fontStyle": "normal",
                    "fontWeight": "700",
                    "letterSpacing": 0,
                    "lineHeight": 45,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Fitness Chat
              </Text>
              <Text
                dataSet={{}}
                style={
                  {
                    "backgroundColor": undefined,
                    "color": "#171717",
                    "fontFamily": undefined,
                    "fontSize": 20,
                    "fontStyle": "normal",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 30,
                    "textDecorationLine": undefined,
                  }
                }
              >
                Be active
              </Text>
            </View>
          </View>
        </View>
      </RNSScreen>
    </RNSScreenStack>
  </View>
</RNCSafeAreaProvider>
`;
